let Keeper;_711‍.x([["addNotes",()=>addNotes],["sendNotes",()=>sendNotes],["deleteNotes",()=>deleteNotes],["getNote",()=>getNote],["editNote",()=>editNote]]);_711‍.w("../models/notes",[["default",["Keeper"],function(v){Keeper=v}]]);

       const addNotes = async (req, res) => {
  const { title, note } = req.body;
  try {
    if (title && note) {
      await new Keeper({
        title: req.body.title,
        body: req.body.note,
        footer: new Date(),
      })
        .save()
        .catch(() => console.log("data save error"));
    } else {
      res.json({ status: "error", message: "Data not store" });
    }
    res.json({ status: "OK" });
  } catch (err) {
    res.json({ status: "error", message: err });
  }
};

       const sendNotes = async (req, res) => {
  try {
    const data = await Keeper.find({});
    res.status(200).json(data);
  } catch (err) {
    res.json({ status: "error", message: err });
  }
};

       const deleteNotes = async (req, res) => {
  try {
    await Keeper.findByIdAndDelete({ _id: req.params.id });
    res.json({ status: "Ok", message: "Deleted Successfully" });
  } catch (err) {
    res.json({ status: "error", message: err });
  }
};

       const getNote = async (req, res) => {
  try {
    const data = await Keeper.findById(req.params.id);
    res.status(200).json(data);
  } catch (err) {
    _711‍.g.console.log(err);
  }
};

       const editNote = async (req, res) => {
  const { title, body } = req.body;
  _711‍.g.console.log(title, body)
  const editNotes = {
    title: titleData,
    body: bodyData,
    footer: new Date(),
  };
  try {
    await Keeper.findByIdAndUpdate(req.params.id, editNotes);
    res.json({ status: "OK" });
  } catch (err) {
    res.json({ status: "error" });
  }
};
